src/python/pants/backend/python/tasks/python_binary_create.py

```diff
@@ -38,6 +38,16 @@ def register_options(cls, register):
                   "created and the command line used to create it. This information may be helpful to you, but means "
                   "that the generated PEX will not be reproducible; that is, future runs of `./pants binary` will not "
                   "create the same byte-for-byte identical .pex files.")
+    # TODO: make an analogy to cls.register_jvm_tool that can be overridden for python subsystems
+    # by a python_requirement_library() target, not just via pants.ini!
+    register('--ptex-injected-pex-version', advanced=True, default='1.6.12', fingerprint=True,
+             help='The pex version to include in the generated .ptex if --generate-ptex is on.')
+    register('--generate-ptex', type=bool, default=False, fingerprint=True,
```

Comment:
Why `ptex`? Does it stand for something I don't know? It's a non-obvious name to me...

Reply:
@jsirois I thought the ski repair analogy was quaint and I loved it -- is there a quick elaboration I can ask of you?

Reply:
Context: https://github.com/pantsbuild/pants/pull/8793/files#discussion_r357070375

Reply:
That link doesn't go anywhere for me...

Reply:
This one doesn't have the `/files` at the end -- i think i may have had a keyboard stutter when copy-pasting the url: https://github.com/pantsbuild/pants/pull/8793#discussion_r357070375

Reply:
Aha... So it's the abbreviation for a kind of plastic that is melted in order to repair skis? That seems like a very oblique reference... How would you feel about a name like bootstrap-pex or self-bootstrapping-pex or similar?

Reply:
If we're opening the discussion, I personally loved just "ipex"!

Reply:
Open to discussion, though! @hrfuller made a great point in a doc comment that it was weird how I tried to avoid giving it a single name.

> self-bootstrapping-pex

I love this!

Reply:
I'm slightly torn on `ipex` - as a user-facing name, I think it's great. Internally, implementation-detail-wise, it's kind of confusing because there are two different pex variants - the code-with-assumed-deps pex and the makes-that-pex-pex, and conflating their names may be confusing. That said, right now the implementation details are actually that the bootstrap pex just makes a full-fat pex.

How about everywhere user facing, we call it an `ipex`, and anywhere that the naming would clash (e.g. the `IPEX-INFO`/`PTEX-INFO` files) we just call it `BOOTSTRAP-PEX-INFO`?

Reply:
I love this approach! Will work to implement this now!!